airsonic-main/src/main/java/org/airsonic/player/service/MediaFileService.java

```diff
@@ -373,14 +374,14 @@ public void populateStarredDate(List<MediaFile> mediaFiles, String username) {
     }
 
     public void populateStarredDate(MediaFile mediaFile, String username) {
-        Date starredDate = mediaFileDao.getMediaFileStarredDate(mediaFile.getId(), username);
+        Instant starredDate = mediaFileDao.getMediaFileStarredDate(mediaFile.getId(), username);
         mediaFile.setStarredDate(starredDate);
     }
 
     private void updateChildren(MediaFile parent) {
 
         // Check timestamps.
-        if (parent.getChildrenLastUpdated().getTime() >= parent.getChanged().getTime()) {
+        if (parent.getChildrenLastUpdated().isAfter(parent.getChanged())) {
```

Comment:
This is a bug. we're not checking the '=' condition, only the '>' condition. Which actually matters a lot.